### PR TITLE
[qt5] MinGW build fixes

### PR DESCRIPTION
**Pull request by koprok**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by FrankXie05 — DISMISSED** [review]

File: ports/qt5-base/patches/mingw9.patch [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+Fixes: QTBUG-94031
+Pick-to: 6.2
+Change-Id: I0b29a4a1932425e1c4079aba6768fe94460c60af
+Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
+Reviewed-by: Oliver Wolff <oliver.wolff@qt.io>
+---
+ src/corelib/io/qfilesystemengine_win.cpp | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
```
Comment (dg0yt) [inline ports/qt5-base/patches/mingw9.patch:18]: In vcpkg, these headers are not desired.

**Review by dg0yt — COMMENTED** [review]
I will test it soon.

**Comment by dg0yt** [comment]
How well did it build for you? I had two overcome two problems, and will send you a PR soon:
- icu libs for x64-mingw-static (a general fix for this port),
- redirect installation from `installed` to `packages` (specific to mingw builds on windows host).

**Comment by koprok** [comment]
@dg0yt, I am building x64-mingw-dynamic triplet. With my changes qt5-base[core] builds without errors and works fine. I have actually never tried the static builds.
I know about the redirection, one of my changes is to fix redirection (in qt_fix_makefile_install.cmake). Doesn't it work for you?

**Comment by dg0yt** [comment]
> I know about the redirection, one of my changes is to fix redirection (in qt_fix_makefile_install.cmake). Doesn't it work for you?

No. The point is that I see the Makefiles generated with `/` path separators (on Windows host, building with mingw32-make from MSYS2), but the fixup is written for `\` path separators. What is in your Makefiles? 
I know that qt scripts change output if `sh` is in `PATH`. We need to handle both cases if you confirm that you see `\`.

**Comment by koprok** [comment]
> What is in your Makefiles? I know that qt scripts change output if `sh` is in `PATH`. We need to handle both cases if you confirm that you see `\`.

Mine are with `\` indeed!
I am using this MinGW: https://github.com/niXman/mingw-builds-binaries/releases/tag/12.2.0-rt_v10-rev0

**Comment by dg0yt** [comment]
You are using a mingw toolchain with less extra tools. `mingw32-make` is for this.
With MSYS2, I use a more complete unixish enviroment. vcpkg already uses msys2 to provide such an environment for building ports based on the autotools build system, using `make`. I would prefer to use the same `make` environment for qt5, to limit variations. We download these packages anyways. But first I want to test if I see a significant difference in build time.

**Comment by koprok** [comment]
> I would prefer to use the same `make` environment for qt5, to limit variations. We download these packages anyways.

Ok, I don't mind as long as msys2's make works with niXman compiler. Please, let me know if I can help with something so that Qt 5 MinGW support gets merged sooner.

**Comment by dg0yt** [comment]
The remaing issue is a little bit tricky, but a serious source of trouble which we must resolve:
- Qt's `configure.bat` calls `mingw32-make`.
- When `sh` is present, Qt generates unixish makefiles with `/`.
- This behaviour is related to `mingw32-make` which uses `sh` instead of `cmd` if present.

I can control all this towards `mingw32-make` with `\`separators. But then then there is the next surprise:
- Qt wants to run a perl script which wants to read from a command which uses `<` for input... 
  Now if msys (unixish) perl is in the `PATH`, it wants to run msys `sh` ... and in this step, the path separators hit the wall again.

So the alternatives are:
- accept both msys and non-msys environments, modify install location accordingly,
- force `\` separators and non-msys perl.
- force `/` separators and msys build.

**Comment by koprok** [comment]
I haven't used msys gcc to build Qt, can you please clarify the following for me to understand better:

> * When `sh` is present, Qt generates unixish makefiles with `/`.

Vcpkg itself does not put `sh` in path, right? So `sh` will be present if the user has `sh` in path beforehand, for example if the user uses msys gcc to build Qt, correct?

> I can control all this towards `mingw32-make` with `\`separators.

How can you do this? How will you remove `sh` from path?

> Now if msys (unixish) perl is in the `PATH`, it wants to run msys `sh` ... and in this step, the path separators hit the wall again.

Again - Vcpkg itself does not put msys perl in path, i.e. msys perl could be in path only if the user has it beforehand, right?

> So the alternatives are:
> 
> * accept both msys and non-msys environments, modify install location accordingly,

Maybe this won't be that difficult. It won't hurt if we try to replace both variants. Only one will get actually replaced.

> * force `\` separators and non-msys perl.

I am not sure if this variant is possible at all, see above questions. How can you (selectively) remove msys `sh` and `perl` from path...

> * force `/` separators and msys build.

Maybe this will be ok too - if Vcpkg itself, intentionally, puts msys `sh` and `perl` in path.

**Comment by dg0yt** [comment]
@FrankXie05 Ready for review.

**Review by FrankXie05 — APPROVED** [review]

**Pull request merged**